mne/io/brainvision/brainvision.py

```diff
@@ -313,57 +317,146 @@ def _read_vmrk_events(fname, event_id=None, trig_shift_by_type=None):
+            onset.append(this_onset)
+            description.append(mtype + '/' + mdesc)
+
+    return np.array(onset), np.array(duration), np.array(description), date_str
+
+
+def _event_id_func(desc, event_id, trig_shift_by_type):
+    """Get integers from string description.
+
+    This function can be passed as event_id to events_from_annotations
+    function.
+
+    Parameters
+    ----------
+    desc : str
+        The description of the event.
+    event_id : dict
+        The default mapping from desc to integer.
+    trig_shift_by_type: dict | None
+        The names of marker types to which an offset should be added.
+        If dict, the keys specify marker types (case is ignored), so that the
+        corresponding value (an integer) will be added to the trigger value of
+        all events of this type. If the value for a key is in the dict is None,
+        all markers of this type will be ignored. If None (default), no offset
+        is added, which may lead to different marker types being mapped to the
+        same event id.
+
+    Returns
+    -------
+    trigger : int | None
+        The integer corresponding to the specific event. If None,
+        then a proper integer cannot be found and the event is typically
+        ignored.
+    """
+    mtype, mdesc = desc.split('/')
+    found = False
+    if mdesc in event_id:
+        trigger = event_id[mdesc]
+        found = True
+    else:
+        try:
+            # Match any three digit marker value (padded with whitespace).
+            # In BrainVision Recorder, the markers sometimes have a prefix
+            # depending on the type, e.g., Stimulus=S, Response=R,
+            # Optical=O, ... Note that any arbitrary stimulus type can be
+            # defined. So we match any single character that is not
+            # forbidden by BrainVision Recorder: [^a-z$%\-@/\\|;,:.\s]
+            marker_regexp = r'^[^a-z$%\-@/\\|;,:.\s]{0,1}([\s\d]{2}\d{1})$'
+            trigger = int(re.findall(marker_regexp, mdesc)[0])
+        except IndexError:
+            trigger = None
+
+    if mtype.lower() in trig_shift_by_type:
+        cur_shift = trig_shift_by_type[mtype.lower()]
+        if cur_shift is not None:
+            trigger += cur_shift
         else:
-            # Markers with no description are not regarded as dropped but
-            # instead are simply ignored. For example, the "New Segment"
-            # markers are ignored, because they usually look like:
-            # Mk1=New Segment,,1,1,0
-            if len(mdesc) > 0:
-                dropped.append(mdesc)
+            # The trigger has been deliberately shifted to None. Do not
+            # add this to "dropped" so we do not warn about something
+            # that was done deliberately. Just continue with next item.
+            trigger = None
+            found = True
+
+    if trigger is None and not found:
+        warn("Marker '%s' will be dropped" % desc)
```

Comment:
The problem of the warnings is here. But reading the tests, that's the expected behavior. The only difference is that the warning message differs from master.